### PR TITLE
I1553 projects 7

### PR DESCRIPTION
**Pull request by Serhiy91**
- Fixed RHS count
- Fixed stuttering

### CODE REVIEW

File: app/imports/client/react/forms/components/EntityArrayForm.js [hunk 1]
```diff
@@ -39,18 +35,24 @@ const EntityArrayForm = ({
...
       const diffs = difference(newItems, items);
       const diffItem = diffs && diffs[0];
       if (!diffItem || (!diffItem.value && !diffItem.label)) return undefined;
 
+      setConfig('keepDirtyOnReinitialize', true);
+      const resetConfig = () => setConfig('keepDirtyOnReinitialize', false);
       if (!diffItem.value) {
-        return onCreate(diffItem).catch(swal.error);
+        return onCreate(diffItem)
+          .then(resetConfig)
```
Comment (Dartv) [inline app/imports/client/react/forms/components/EntityArrayForm.js]: I think it's better to use `finally` there

File: app/imports/client/react/forms/components/EntityArrayForm.js [hunk 2]
```diff
@@ -39,18 +35,24 @@ const EntityArrayForm = ({
...
+          .catch(swal.error);
       }
-      return onUpdate(diffItem).catch((error) => {
-        swal.error(error);
-        reset();
-      });
+      return onUpdate(diffItem)
+        .then(resetConfig)
```
Comment (Dartv) [inline app/imports/client/react/forms/components/EntityArrayForm.js]: Why do you need it `onUpdate`?

Comment (Serhiy91) [reply app/imports/client/react/forms/components/EntityArrayForm.js]: I need it to fix stuttering while updating

Comment (Dartv) [reply app/imports/client/react/forms/components/EntityArrayForm.js]: Why is there stuttering while updating? There shouldn't be

File: app/imports/client/react/organization-settings/components/DepartmentsSubcardForm.js [hunk 3]
```diff
@@ -20,17 +20,31 @@ const DepartmentsSubcardForm = ({ organizationId, isOpen }) => (
...
+        mutation={Mutations.CREATE_DEPARTMENT}
+        refetchQueries={[
+          { query: Queries.DEPARTMENT_LIST, variables: { organizationId } },
+          { query: Queries.DEPARTMENT_COUNT, variables: { organizationId } },
+        ]}
+        children={noop}
+      />,
+      <Mutation
+        mutation={Mutations.DELETE_DEPARTMENT}
+        refetchQueries={[
```
Comment (Dartv) [inline app/imports/client/react/organization-settings/components/DepartmentsSubcardForm.js]: `refetchQueries` passed to `Mutation` component should be a function

File: app/imports/client/react/organization-settings/components/ProjectsSubcardForm.js [hunk 4]
```diff
@@ -20,17 +20,31 @@ const ProjectsSubcardForm = ({ organizationId, isOpen }) => (
         onError={swal.error}
         children={noop}
       />,
-      <Mutation mutation={Mutations.CREATE_PROJECT} children={noop} />,
+      <Mutation
+        mutation={Mutations.CREATE_PROJECT}
+        refetchQueries={[
```
Comment (Dartv) [inline app/imports/client/react/organization-settings/components/ProjectsSubcardForm.js]: same there

File: app/imports/client/react/forms/components/EntityArrayForm.js (hunk 2 above)
Comment (Serhiy91) [reply app/imports/client/react/forms/components/EntityArrayForm.js]: The same cause as for creating. First rerendering with old data (before Mutation is completed).

**Pull request merged**